kratos/utilities/amgcl_csr_conversion_utilities.h

```diff
@@ -0,0 +1,82 @@
+
+//    |  /           | 
+//    ' /   __| _` | __|  _ \   __| 
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/ 
+//                   Multi-Physics  
+//
+//  License:		 BSD License 
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Denis Demidov
+//                   Riccardo Rossi
+//                   
+//
+
+#if !defined(KRATOS_CSR_CONVERSION_UTILITIES_H_INCLUDED)
+#define  KRATOS_CSR_CONVERSION_UTILITIES_H_INCLUDED
+
+#include "containers/csr_matrix.h"
+
+#include <amgcl/backend/builtin.hpp>
+#include <amgcl/adapter/zero_copy.hpp>
+#include <span/span.hpp>
+
+namespace Kratos
+{
+
+/**
+Utilities to convert the distributed_csr matrix to other libraries
+ */
+class AmgclCSRConversionUtilities
+{
+
+public:
+
+    /**
+     This function returns a shared pointer to an Amgcl distributed_matrix
+     */
+	template< class TDataType, class TIndexType >
+	static Kratos::shared_ptr<typename amgcl::backend::builtin<TDataType>::matrix > ConvertToAmgcl(
+			const CsrMatrix<TDataType, TIndexType>& rA)
+	{
+    	Kratos::shared_ptr<typename amgcl::backend::builtin<TDataType>::matrix > pAmgcl = amgcl::adapter::zero_copy(
+                rA.size1(),
+                rA.index1_data().begin(),
+                rA.index2_data().begin(),
+                rA.value_data().begin()
+            );
+        pAmgcl->own_data = false;
+        return pAmgcl;
+	}
+
+    template< class TDataType, class TIndexType >
+	static CsrMatrix<TDataType, TIndexType> ConvertToCsrMatrix(
+			typename amgcl::backend::builtin<TDataType>::matrix& rA
+			)
+	{
+        CsrMatrix<TDataType, TIndexType> Aconverted;
+
+        if(rA.own_data == false){ //if rA is not the owner, Aconverted cannot be
+            Aconverted.SetIsOwnerOfData(false);
+        }
+        else{ //if rA is the owner, transfer ownership to the csr_matrix
+            rA.own_data = false;
+            Aconverted.SetIsOwnerOfData(true);
+        }
+
+        Aconverted.SetRowSize(rA.nrows);
+        Aconverted.SetColSize(rA.ncols);
+        Aconverted.AssignIndex1Data((TIndexType*)(rA.ptr), rA.nrows+1);
+        Aconverted.AssignIndex2Data((TIndexType*)(rA.col), rA.nnz);
+        Aconverted.AssignValueData((TDataType*)(rA.val), rA.nnz);
+        return Aconverted;
```

Comment:
no `std::move` here?
You use it in the MPI-equivalent

Reply:
well, i get the following if i add it

~~~
In file included from /mnt/c/Users/rross/Programming/Kratos/kratos/mpi/utilities/amgcl_distributed_csr_conversion_utilities.h:24,
                 from /mnt/c/Users/rross/Programming/Kratos/kratos/mpi/tests/sources/test_distributed_sparse_graph.cpp:28:
/mnt/c/Users/rross/Programming/Kratos/kratos/utilities/amgcl_csr_conversion_utilities.h: In instantiation of ‘static Kratos::CsrMatrix<TDataType, TIndexType> Kratos::AmgclCSRConversionUtilities::ConvertToCsrMatrix(typename amgcl::backend::builtin<T1>::matrix&) [with TDataType = double; TIndexType = long unsigned int; typename amgcl::backend::builtin<T1>::matrix = amgcl::backend::crs<double, long int, long int>]’:
/mnt/c/Users/rross/Programming/Kratos/kratos/mpi/utilities/amgcl_distributed_csr_conversion_utilities.h:113:114:   required from ‘static Kratos::DistributedCsrMatrix<TDataType, TIndexType> Kratos::AmgclDistributedCSRConversionUtilities::ConvertToCsrMatrix(amgcl::mpi::distributed_matrix<amgcl::backend::builtin<double> >&) [with TDataType = double; TIndexType = long unsigned int]’
/mnt/c/Users/rross/Programming/Kratos/kratos/mpi/tests/sources/test_distributed_sparse_graph.cpp:493:109:   required from here
/mnt/c/Users/rross/Programming/Kratos/kratos/utilities/amgcl_csr_conversion_utilities.h:73:36: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
   73 |         return std::move(Aconverted);
      |                                    ^
/mnt/c/Users/rross/Programming/Kratos/kratos/utilities/amgcl_csr_conversion_utilities.h:73:36: note: remove ‘std::move’ call
~~~